doc(k4k8s) add kong-enterprise-edition DB-less doc (#2191)
Kong/docs.konghq.com: app/enterprise/2.1.x/kong-for-kubernetes/deployment-options.md

skip_read_time: true
---
<div class="alert alert-ee blue">

> For the {{site.ee_product_name}} 2.1.x beta, Kong for Kubernetes Enterprise

Reviewer: Is this just a temporary change for beta, or are you fully stopping development of the kong-enterprise-k8s image going forward?

Author: Pending customer feedback from the beta, but as of now, assuming no unexpected issues using the `kong-enterprise-edition` image, yes, we will end updates for `kong-enterprise-k8s` and direct existing users to switch images for the upgrade to 2.1 GA.

> (K4K8s) uses the <code>kong-enterprise-edition</code> image, which works as a

Reviewer: Are we still keeping the same naming convention, i.e. K4K8s for DB-less and Kong Enterprise on Kubernetes for DB-backed?

Author: Per Reza's comment in DOCS-1166, effectively yes for now. It's not ideal but it will suffice for the beta; it's an area that we should expect user confusion on and we should revisit the naming questions before GA.

_These repositories require a login. If you see a 404, log in through the [Kong
repository home page](https://bintray.com/kong) first._

As of 2.1.x, the `kong-enterprise-edition` image supports DB-less operation and
is recommended for all deployments. Existing users of the `kong-enterprise-k8s`
image wishing to try the 2.1.x beta should switch to the
`kong-enterprise-edition` image.

> Switching images should not require configuration changes other than changing
> the image itself and creating a registry secret for `kong-enterprise-edition`
> if you have not created one already. If you encounter issues after switching
> images, please [contact Kong Enterprise Support][support].

Reviewer: ```suggestion
Switching images requires the following configuration changes:
* Changing the image to `kong-enterprise-edition`.
* Creating a registry secret for `kong-enterprise-edition`, if you have not created one already.
If you encounter issues after switching images, please 
[contact Kong Enterprise Support][support].
```
I'd avoid "should", too uncertain.

Use the `kong-enterprise-edition` image and a database-backed deployment
if you want to use:

* Dev Portal
* Brain
* Immunity
* Teams (RBAC)
* Workspaces

In the 2.1.x beta, the following features have support in DB-less mode, but
work differently than in DB-backed modes:

* Kong Manager (read-only)
* Vitals (using [Prometheus][vitals-prometheus] or [InfluxDB][vitals-influxdb]
  strategies)

Because Kong for Kubernetes is configured by the ingress controller, some
functionality in these features is different from traditional deployments:

> * Instead of using Kong Manager, proxy configuration is normally managed by the

Reviewer: ```suggestion
* Instead of using Kong Manager, proxy configuration is managed by the
```
